tests/test_cli.py

```diff
@@ -304,8 +304,10 @@ def test_option_hierarchy():
         catch_exceptions=False
     )
 
-    expected = "Product 0e66b563-78d9-4480-9c3d-b64a60cf1a9f - Date: 2016-12-01T14:10:42Z, Instrument: MSI, Mode: , Satellite: Sentinel-2, Size: 526.15 MB"
-    assert re.findall("^Product .+$", result.output, re.M)[1] == expected
+    products = re.findall("^Product .+$", result.output, re.M)
+    # Check that all returned products are of type 'S2MSI1C'
+    assert len(products) > 0
+    assert all("Instrument: MSI, Mode: , Satellite: Sentinel-2" in p for p in products)
```

Comment:
The logic is still sound and you improved the test by asserting that all products are correct, not just the first one. Great job!